src/interface/update.hpp

```diff
@@ -74,7 +75,9 @@ TaskStatus WeightedSumData(const std::vector<F> &flags, T *in1, T *in2, const Re
       DEFAULT_LOOP_PATTERN, "WeightedSumData", DevExecSpace(), 0, x.GetDim(5) - 1, 0,
       x.GetDim(4) - 1, 0, x.GetDim(3) - 1, 0, x.GetDim(2) - 1, 0, x.GetDim(1) - 1,
       KOKKOS_LAMBDA(const int b, const int l, const int k, const int j, const int i) {
-        z(b, l, k, j, i) = w1 * x(b, l, k, j, i) + w2 * y(b, l, k, j, i);
+        if (x.IsAllocated(b, l) && y.IsAllocated(b, l) && z.IsAllocated(b, l)) {
```

Comment:
I agree that some more thought will be necessary to figure out exactly how to handle this. And it may only become clear once we have downstream codes that make extensive use of the sparse capability. Until then, I think it's best to make the least amount of changes to make true sparse work while keeping results the same.